Makefile

```diff
@@ -22,11 +22,13 @@ GOBINPATH    := $(shell $(GO) env GOPATH)/bin
 VERSION      := $(shell cat VERSION)
 COMMIT       := $(shell git rev-parse --short HEAD 2>/dev/null)
 BUILD_DATE   := $(shell date +%Y%m%d)
+TAG          := $(shell git show-ref --tags | grep $(COMMIT) | awk '{print $2}' | cut -d/ -f3)
```

Comment:
This should be `git describe --tags` I think because 
```
nirmoy@chapman:~/go/src/github.com/SUSE/skuba> git show-ref --tags
c30833c271fdbd935ed159969032fa4715382395 refs/tags/v0.1.0
96a14d2d2c12a08871f25773feaf392dc5a701ca refs/tags/v0.2.0
d4343eb8035b3adafcb34604114f3f28d24a46dc refs/tags/v0.2.1
f55963df90dab3bfd33c977f239c7db31820e011 refs/tags/v0.3.0
eed514f2e5c868ea66b7eeef3442392ae628010e refs/tags/v0.3.1
d3698ec303c0e80f27dfefe4acb1d9a2ff6e3c20 refs/tags/v0.3.1.1
dd96fed5c766de8c3c845b23f761fdf41b4f4030 refs/tags/v0.4.0
67e823908db569d2992fc5bc1907663d21e2e393 refs/tags/v0.5.0
e3b493b0151fbbe5349cb10fb2a8b58160002b26 refs/tags/v1.4.0
851cca8e52c49fcb5e20bfb17e3e3b3611e6a6f9 refs/tags/v1.4.1
nirmoy@chapman:~/go/src/github.com/SUSE/skuba> git describe --tags
v1.4.1
nirmoy@chapman:~/go/src/github.com/SUSE/skuba> make
GO111MODULE=on go install -mod=vendor -ldflags "-X=github.com/SUSE/skuba/pkg/skuba.Version= -X=github.com/SUSE/skuba/pkg/skuba.Commit=1dd16815 -X=github.com/SUSE/skuba/pkg/skuba.BuildDate=20190813 -X=github.com/SUSE/skuba/pkg/skuba.Tag=" -tags development ./cmd/...
rm -f /home/nirmoy/go/bin/kubectl-caasp
ln -s /home/nirmoy/go/bin/skuba /home/nirmoy/go/bin/kubectl-caasp
nirmoy@chapman:~/go/src/github.com/SUSE/skuba> skuba version 
** This is an UNTAGGED version and NOT intended for production usage. **
skuba version: untagged (development) 1dd16815 20190813 go1.12.6

```

Reply:
I think you didn't tag a new version on that specific commit, that's why it's untagged. If you try the PR, and then:

`~/p/g/s/g/S/skuba (dynamic-version-warning) > git tag v0.1.0-alpha2
~/p/g/s/g/S/skuba (dynamic-version-warning) > make
GO111MODULE=on go install -mod=vendor -ldflags "-X=github.com/SUSE/skuba/pkg/skuba.Version=0.1.0 -X=github.com/SUSE/skuba/pkg/skuba.Commit=44589957 -X=github.com/SUSE/skuba/pkg/skuba.BuildDate=20190813 -X=github.com/SUSE/skuba/pkg/skuba.Tag=v0.1.0-alpha2" -tags development ./cmd/...
rm -f /home/ereslibre/projects/go/bin/kubectl-caasp
ln -s /home/ereslibre/projects/go/bin/skuba /home/ereslibre/projects/go/bin/kubectl-caasp
~/p/g/s/g/S/skuba (dynamic-version-warning) > skuba version
** This is an ALPHA release and NOT intended for production usage. **
skuba version: 0.1.0 (development) 44589957 (tagged as "v0.1.0-alpha2") 20190813 go1.12.7
`

It should work with any kind of tag, like `v0.1.0-alpha`, `v0.1.0-beta`, `v0.1.0-rc`, `v0.1.0`. Without a tag on the current commit, it will assume it's an untagged version and report it.

`git describe --tag` always returns a tag, but we only want to report as tagged commits that are really tagged I'd say.

Reply:
Thank you @nirmoy, updated the PR so if the current HEAD is not a tag we will use the `git describe --tag` output instead, like this:

```
~/p/g/s/g/S/skuba (dynamic-version-warning) > skuba version
** This is an UNTAGGED version and NOT intended for production usage. **
skuba version: untagged (v0.9.4-17-g1602d982) (development) 20190814 go1.12.7
```